### PR TITLE
Sprint and ingame base state

### PR DESCRIPTION
**Pull request by matcool**
finally added the sprint gamemode, although currently times aren't saved anywhere

### CODE REVIEW

File: src/states/sprint.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	if (piece.dead()) {
+		game.pushState(make_unique<Results>(this));
+		return;
+	}
+
+	time += dt;
+
+	infoText.setText(sdscatprintf(sdsempty(), "lines: %d\ntime: %.3f",
```
Comment (geniiii) [inline src/states/sprint.cpp]: lines -> Lines  
same for time

File: src/board.cpp [hunk 2]
```diff
@@ -1,7 +1,13 @@
 #include <3dstris/board.hpp>
 
 Board::Board(u32 width, u32 height) : width(width), height(height) {
-	grid.resize(width * height, PieceType::None);
+    reset();
+}
+
+void Board::reset() {
+    _linesCleared = 0;
+    grid.resize(0);
```
Comment (geniiii) [inline src/board.cpp]: discussed on discord

File: include/3dstris/states/playing.hpp [hunk 3]
```diff
@@ -2,31 +2,14 @@
 
 #include <3dstris/board.hpp>
 #include <3dstris/shapes.hpp>
```
Comment (geniiii) [inline include/3dstris/states/playing.hpp]: these aren't necessary anymore

File: include/3dstris/states/playing.hpp [hunk 4]
```diff
@@ -2,31 +2,14 @@
...
-#include <3dstris/state.hpp>
+#include <3dstris/ingame.hpp>
 #include <3dstris/util.hpp>
 #include <deque>
 #include <random>
```
Comment (geniiii) [inline include/3dstris/states/playing.hpp]: these aren't necessary anymore

File: include/3dstris/states/sprint.hpp [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <3dstris/board.hpp>
+#include <3dstris/ingame.hpp>
+#include <3dstris/shapes.hpp>
+#include <3dstris/util.hpp>
+#include <3dstris/util/text.hpp>
+#include <deque>
+#include <random>
```
Comment (geniiii) [inline include/3dstris/states/sprint.hpp]: clean these up too

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -43,8 +46,11 @@ add_library(3dstris STATIC
 	game.cpp
 	config.cpp
 	state.cpp
+        ingame.cpp
 	states/mainmenu.cpp
-	states/playing.cpp
+        states/modeselect.cpp
+        states/playing.cpp
+        states/sprint.cpp
```
Comment (geniiii) [inline src/CMakeLists.txt]: wtf happened to the formatting

**Comment by matcool** [comment]
> lgtm

sdfs

**Comment by geniiii** [comment]
> > lgtm
> 
> sdfs

me too

**Pull request merged**